controllers/cards.js

```diff
@@ -22,27 +18,36 @@ module.exports.createCard = (req, res) => {
   const { name, link } = req.body;
   cardModel.create({ name, link, owner: req.user._id })
     .then((card) => res.status(200).send({ data: card }))
-    .catch((err) => ((err.name === 'ValidationError') ? res.status(400).send({ message: 'Ошибка валидации' }) : res.status(500).send({ message: 'Произошла ошибка' })));
+    .catch((err) => res.status(500).send({ message: err.message }));
 };
 
-
 //  удаляет карточку по идентификатору
 module.exports.deleteCard = (req, res) => {
   const { cardId } = req.params;
+  const ownerId = req.user._id;
 
   if (!ObjectId.isValid(cardId)) {
-    res.status(400).send({ message: 'Невалидный id' });
-    return;
+    throw new BadRequestError('Невалидный id');
   }
+
   cardModel.findById(cardId)
-    .orFail(() => new NotFoundError('Карточка не найдена'))
     .then((card) => {
-      if (!card.owner.equals(req.user._id)) {
-        res.status(403).send({ message: 'Отсутствует доступ' });
-      } else {
-        cardModel.findByIdAndRemove(cardId)
-          .then(() => res.status(200).send({ data: card }));
+      if (!card) {
+        throw new NotFoundError('Нет карточки с таким id');
+      }
+      return card.owner.equals(ownerId);
+    })
+    .then((isMatch) => {
+      if (!isMatch) {
+        throw new ForbidError('Невозможно удалить чужую карточку');
+      }
+      return cardModel.findByIdAndRemove(cardId);
+    })
+    .then((cardRemove) => {
+      if (!cardRemove) {
+        throw new ForbidError('Невозможно удалить чужую карточку');
       }
+      res.send({ remove: cardRemove });
     })
-    .catch((err) => res.status(err.statusCode || 500).send({ message: 'Что-то пошло не так' }));
+    .catch((err) => res.status(err.statusCode).send({ message: err.message }));
```

Comment:
Можно лучше: использовать централизованный обработчик ошибок для отправки ответа.
Стало актуально после того, как обработчик ошибок был внедрён и он начал рабоать. В целом это вопрос следующей практики, но раз часть реализована, можно внедрить.

В концепции _Express_ можно задать сколько угодно промежуточных обработчиков, передача управления следующему обработчику осуществляется через вызов функции `next()`. В обычных обработчиках это 3-й аргумент функции, в обработчике ошибок 4-й (тк первый становится `err`): https://expressjs.com/ru/guide/using-middleware.html .

Чтобы ошибка была подхвачена обработчиком ошибок достаточно передать `err` в `next()`:
```suggestion
    .catch(next);
```
_с учётом, что `next` уже объявлена_

После этого обработчик ошибок определит код ошибки используя `.statusCode` и произведет отправку корректного ответа:
https://github.com/YuriyFeo/project14.github.io/blob/3e4852c1ad4e3160a84dbff6200fb46bee0e2fe9/app.js#L39-L46

В случае, когда ожидается ответ отличный от 500, нужно следовать примеру с созданием и использованием отдельного класса ошибки с HTTP кодом.